tests/utils/__init__.py

```diff
@@ -41,3 +43,18 @@ def _parse_times(a_dict: Dict[str, Any]) -> None:
     _parse_times(d1)
     _parse_times(d2)
     test_class.assertDictEqual(d1, d2)
+
+
+# Mypy raises an error for this class:
+#  error: Missing type parameters for generic type "TemporaryDirectory"  [type-arg]
+# Trying to add a concrete type (e.g. TemporaryDirectory[str]) satisfies mypy, but
+# raises a runtime exception:
+#  TypeError: 'type' object is not subscriptable
+class TemporaryDirectory(tempfile.TemporaryDirectory):  # type: ignore [type-arg]
+    def __init__(self, suffix: Optional[str] = None, prefix: Optional[str] = None):
+        """In the GitHub Actions Windows runner the default TMPDIR directory is on a
+        different drive (C:\\) than the code and test data files (D:\\). This was causing
+        failures in os.path.relpath on Windows, so we put the temp directories in the
+        current working directory instead. There os a "tmp*" line in the .gitignore file
+        that ignores these directories."""
+        super().__init__(suffix=suffix, prefix=prefix, dir=os.getcwd())
```

Comment:
@l0b0 @lossyrob I couldn't find a way around ignoring this issue (see the comment for the error I was running into). If anyone has an idea for fixing this the right way I'm happy to include it.

Reply:
You could define a method that returns a `TemporaryDirectory` instance e.g.

```python
import os
from tempfile import TemporaryDirectory

def get_tmp_dir() -> TemporaryDirectory:
  return TemporaryDirectory(dir=os.getcwd())

with get_tmp_dir() as tmp_dir:
  print(tmp_dir)
```

Reply:
Yeah, I tried that first, but it ran into the same typing issue.

Reply:
Huh, very strange. I would say using the method for simplicity and adding `# type:ignore` with a note about why is the best workaround.

Reply:
Nvm, I was able to fix the typing issue with help from [this comment](https://github.com/python/mypy/issues/5264#issuecomment-399407428). Apparently it's a known issue with Python standard library classes.

Reply:
Replaced the sub-class with a function and fixed the type issue in d23be0a2dc2459d308eb332a2fda849936f2cfce